docs/tutorials/custom_experiment.rst

```diff
@@ -0,0 +1,362 @@
+    ):
+        """Basic randomize Z-basis measurement via a Pauli frame transformation
+        
+        Note this will just append a new set of measurment at the end of a circuit.
+        A more advanced version of this experiment would be use a transpiler pass to
+        replace all exisiting measurements in a circuit with randomized measurements.
+        """
+        if physical_qubits is None:
+            physical_qubits = tuple(range(circuit.num_qubits))
+        if measured_qubits is None:
+            measured_qubits = tuple(range(circuit.num_qubits))
+        
+        # Initialize BaseExperiment
+        analysis = RandomizedMeasurementAnalysis()
+        super().__init__(physical_qubits, analysis=analysis, backend=backend)
+        
+        # Add experiment properties
+        self._circuit = circuit        
+        self._measured_qubits = measured_qubits
+        
+        # Set any init optinos
+        self.set_experiment_options(**experiment_options)
+
+    @classmethod
+    def _default_experiment_options(cls):
+        options = super()._default_experiment_options()
+        options.num_samples = "default"
+        options.seed = None
+        return options
+    
+    def circuits(self):
+        # Number of classical bits for original circuit and added measurements
+        circ_nc = self._circuit.num_clbits
+        meas_nc = len(self._measured_qubits)
+        circ_qubits = list(range(self.num_qubits))
+        circ_clbits = list(range(circ_nc))
+        meas_qubits = self._measured_qubits
+        meas_clbits = list(range(circ_nc, circ_nc + meas_nc))
+
+        # Get number of samples from options
+        num_samples = self.experiment_options.num_samples
+        if num_samples == "default":
+            num_samples = 2 ** self.num_qubits
+        
+        # Get rng seed
+        seed = self.experiment_options.seed
+        if isinstance(seed, Generator):
+            rng = seed
+        else:
+            rng = default_rng(seed)
+
+        # Sample Paulis this might have duplicates, but we don't really
+        # have any easy way of running different number of shots per circuit
+        # so we just run repeat circuits multiple times
+        paulis = random_pauli_list(meas_nc, size=num_samples, phase=False, seed=rng)
+
+        # Construct circuits
+        circuits = []
+        orig_metadata = self._circuit.metadata or {}
+        for pauli in paulis:
+            name = f"{self._circuit.name}_{str(pauli)}"
+            circ = QuantumCircuit(
+                self.num_qubits, circ_nc + meas_nc,
+                name=name
+            )
+            # Append original circuit
+            circ.compose(
+                self._circuit, circ_qubits, circ_clbits, inplace=True
+            )
+
+            # Add Pauli frame
+            circ.compose(pauli, meas_qubits, inplace=True)
+
+            # Add final Measurement
+            circ.measure(meas_qubits, meas_clbits)
+    
+            # Add metadata
```

Comment:
Explain the role of the metadata. Explain that the analysis is going to use it, this is a way to communicate between the experiment and its analysis.